### ISSUE TITLE
Provide some data samples for nerf pretraining

### ISSUE THREAD

**Issue by tunglm2203** [jayLEE0301/snerl_official#1]
Hi, thank you for publishing your code. This method is very useful to learn 3D features for RL. 

I am trying to prepare the data for the pre-training phase, but I'm not sure about its format.

Could you provide some data samples such as  JSON file (contains 'transform_matrix', 'camera_angle_x', etc.), semantic image, and RGB image from 3 views?

I really appreciate it if you can share this!
Thank you.

**Comment by jayLEE0301** [comment]
The cameras we used are as follows:
cam_1_1, cam_7_4, cam_14_2

In the notation _cam_i_j_, i and j indicate each camera's Euler angles. The Euler angles for cam_i_j can be obtained using the equation below:

euler_x = 2 * np.pi * i / LONGITUDE
euler_y = - np.pi * (j + 0.5) / (2 * LATITUDE)

We have also added various other cameras to the XML file, which can be found in the following location (we attach the add_cam.py below to clarify how we add the cameras to metaworld XML files.):
metaworld/metaworld/envs/assets_v2/objects/assets/xyz_base.xml

Thus, you can calculate the camera matrix based on the distance, and Euler angles of the camera. (or you can use the functions is snerl/encoder.py as follows)
```
trans_t = lambda t : torch.Tensor([
    [1,0,0,0],
    [0,1,0,0],
    [0,0,1,t],
    [0,0,0,1]]).float()

rot_phi = lambda phi : torch.Tensor([
    [1,0,0,0],
    [0,np.cos(phi),-np.sin(phi),0],
    [0,np.sin(phi), np.cos(phi),0],
    [0,0,0,1]]).float()

rot_theta = lambda th : torch.Tensor([
    [np.cos(th),0,-np.sin(th),0],
    [0,1,0,0],
    [np.sin(th),0, np.cos(th),0],
    [0,0,0,1]]).float()


def pose_spherical(theta, phi, radius):
    c2w = trans_t(radius)
    c2w = rot_phi(phi/180.*np.pi) @ c2w
    c2w = rot_theta(theta/180.*np.pi) @ c2w
    c2w = torch.Tensor(np.array([[-1,0,0,0],[0,0,1,0],[0,1,0,0],[0,0,0,1]])) @ c2w
    return c2w
```



#### add_cam.py
```
import os
import functools

import numpy as np
from scipy.spatial.transform import Rotation as R
import copy
import matplotlib.pyplot as plt
import cv2
import xml.etree.ElementTree as ET

from tests.metaworld.envs.mujoco.sawyer_xyz.test_scripted_policies import ALL_ENVS, test_cases_latest_nonoise


LONGITUDE = 20
LATITUDE = 5

def add_cam():
    xml_path = os.path.join("metaworld/metaworld/envs/assets_v2/objects/assets/xyz_base.xml")
    tree = ET.parse(xml_path)
    worldbody = tree.getroot()
    dist = 0.5
    for i in range(LONGITUDE):
        for j in range(LATITUDE):
            euler_x = 2 * np.pi * i / LONGITUDE
            euler_y = - np.pi * (j + 0.5) / (2 * LATITUDE)
            pos_z = dist * np.cos(-euler_y)
            pos_x = dist * np.sin(-euler_y) * np.sin(euler_x)
            pos_y = dist * np.sin(-euler_y) * np.cos(euler_x)
            
            
            r = R.from_euler('xyz', [-np.pi + euler_x, 0, euler_y])
            
            ET.SubElement(
                      worldbody, "camera",
                      name="cam_%d_%d" % (i, j),
                      mode="fixed",
                      fovy="90",
                      pos="%f %f %f" % (pos_x, pos_y, pos_z),
                      quat="%f %f %f %f" % (r.as_quat()[0], r.as_quat()[1], r.as_quat()[2], r.as_quat()[3]),
                  )
    tree.write("metaworld/metaworld/envs/assets_v2/objects/assets/camera_tree.xml")
    
    
add_cam()
```

If you have any additional questions, please let us know.
Thank you.

**Comment by tunglm2203** [comment]
Thank you @jayLEE0301. 

I was able to generate information for NERF training. However, for semantic data, I found that all image's values are divided by 50 in [(here)](https://github.com/jayLEE0301/snerl_official/blob/7eadf08f73fefaa89dac33da62a4e052c2486d13/nerf_pretrain/load_blender.py#L66). What is the reason for this?

Code for loading sematic label **(load_blender.py):**
```
semantics.append(np.array(Image.open(os.path.join(basedir,  frame['file_path'][:14] + 'semantic_' + frame['file_path'][14:]+ '.png'))).astype(np.float32)/50)
```

Additionally, your semantic linear head outputs [(5 classes)](https://github.com/jayLEE0301/snerl_official/blob/7eadf08f73fefaa89dac33da62a4e052c2486d13/nerf_pretrain/run_nerf_helpers.py#L94). Does this mean that you only consider 5 classes for semantic prediction?

Code for defining sematic head **(run_nerf_helpers.py):**
```
self.semantic_linear = nn.Sequential(nn.Linear(W,W), nn.Linear(W, 5))
```

**Comment by dsshim0125** [comment]
1. The reason for dividing image value by 50 is that we saved the semantic label of the image by multiplying 50 ( It means that in our saved semantic label file, the label 1 is saved as 50 and 3 saved as 150)
2. We only consider 5 classes in the window environment and the number of the class is dependent on the environment. You can change the number of the outputs of the semantic branch with different number of class labels.

**Comment by jayLEE0301** [comment]
I think the issue has been addressed, so I'll close it.
Thank you.
